File: app/src/main/java/com/droidapp/Main2Activity.java
package com.droidapp;

import android.os.Bundle;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import com.droidnet.old.DroidListener;
import com.droidnet.old.DroidNet;


public class Main2Activity extends AppCompatActivity implements DroidListener {

    private TextView mTvStatus;
    private DroidNet mDroidNet;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main2);

        mTvStatus = findViewById(R.id.tv_status);

        mDroidNet = DroidNet.getInstance();
        mDroidNet.addInternetConnectivityListener(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mDroidNet.removeInternetConnectivityChangeListener(this);
    }

    @Override
    public void onInternetConnectivityChanged(boolean isConnected) {
        if (isConnected) {
            mTvStatus.setText("connected");
        } else {
            mTvStatus.setText("not connected");
        }
    }
}

File: app/src/main/java/com/droidapp/Main2Activity.kt
package com.droidapp

import android.os.Bundle
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.Observer
import com.droidnet.new.Event
import com.droidnet.new.NetworkEvents
import com.droidnet.new.NetworkState
import com.droidnet.new.NetworkStateHolder

class Main2Activity : AppCompatActivity() {
    private var mTvStatus: TextView? = null
    private var previousSate = true

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main2)
        mTvStatus = findViewById(R.id.tv_status)


        NetworkEvents.observe(this, Observer {
            if (it is Event.ConnectivityEvent)
                handleConnectivityChange(it.state)
        })
    }

    private fun handleConnectivityChange(networkState: NetworkState) {
        if (networkState.isConnected && !previousSate) {
            mTvStatus!!.text = "connected"
        }

        if (!networkState.isConnected && previousSate) {
            mTvStatus!!.text = "not connected"
        }

        previousSate = networkState.isConnected
    }

    override fun onResume() {
        super.onResume()
        handleConnectivityChange(NetworkStateHolder)
    }
}